convert2rhel/main.py

```diff
@@ -63,16 +64,39 @@ def initialize_logger(log_name, log_dir):
 
 
 def main():
-    """Perform all steps for the entire conversion process."""
+    """
+    Wrapper around the main entrypoint.
+
+    Performs everything necessary to set up before starting the actual
+    conversion process itself, then calls main_locked(), protected by
+    the application lock, to do the conversion process.
+    """
 
     process_phase = ConversionPhase.INIT
 
+    # Make sure we're being run by root
+    utils.require_root()
+
     # initialize logging
     initialize_logger("convert2rhel.log", logger_module.LOG_DIR)
 
     # handle command line arguments
     toolopts.CLI()
 
+    try:
+        with applock.ApplicationLock("convert2rhel"):
+            return main_locked(process_phase)
+    except applock.ApplicationLockedError:
+        # We have not rotated the log files at this point because main.initialize_logger()
+        # has not yet been called.  So we use sys.stderr.write() instead of loggerinst.error()
+        sys.stderr.write("Another copy of convert2rhel is running.\n")
+        sys.stderr.write("\nNo changes were made to the system.\n")
```

Comment:
Oops, we are rotating the logs already: on line 81.  No worries, though.  I'm not sure how the PR to fix --help for non-privileged users will change the position of these function calls. We can switch this to use loggerinst if it's still appropriate after that PR is coded.